### ISSUE TITLE
Touch with Assistive Technology

### ISSUE THREAD

**Issue by awkawk** [w3c/wcag21#157]
Scott Hollier (http://hollier.info/wcag21draft/):
...the standout point for me is touch with assistive technology. 

 Its remarkable how often aps work great before I turn on the screen reader on my phone, and how completely inaccessible it becomes once the screen reader is enabled.  While the other SC are quite specific, it’s this broad requirement of AT compatibility that I suspect will be one of the greatest arguments put forward for moving to WCAG 2.1 and it’s my hope that this is adopted by the AG WG as soon as possible.

**Comment by awkawk** [comment]
This issue was addressed in combination with two other SC proposals with a conformance note: 
A full page includes each variation of the page that is automatically generated by the page for various screen sizes. Each of these variations needs to conform (or needs to have a conforming alternate version) in order for the entire page to conform.

**Comment by detlevhfischer** [comment]
Hi Scott Hollier,
The discussions around this proposed SC have shown that due to the differences between browsers and AT there may be no conclusive way to test if content meets this criterion.

Also, the requirement seemed mostly covered by 2.1.1 Keyboard in the sense that if a control can be focused and activated with a keyboard, presumably it would also be possible to swipe to it and activate it via double tap it in a mobile screenreader scenario.

Another point is that the issue is also not exclusively related to touch, other input methods may be used concurrenty while AT is turned on. This is the topic in the new [SC 2.5.2 Concurrent Input
Mechanisms](https://www.w3.org/TR/2017/WD-WCAG21-20170912/#concurrent-input-mechanisms).

Another new success criterion, [2.5.1 Pointer Gestures](https://www.w3.org/TR/2017/WD-WCAG21-20170912/#pointer-gestures) is meant to close the gap that exists in situations where touch is the input mode and no physical or virtual keyboard alternative is present (a virtual keyboard may not be shown on iOS unless the focus is on some text input). It requires that operation should be possible via single untimed pointer gestures  - i.e. author-defined gestures (swiping to scroll or load content in infinite scroll scenarios, etc) that may not work when AT is turned on would need a fallback way of operation.

A [new note under Conformance requirement 1 Full pages](https://www.w3.org/TR/2017/WD-WCAG21-20170912/#h-note29) explains that a full page includes each variation of the page that is automatically generated by the page for various screen sizes. This would, for example, cover situations where controls in responsive views on narrow viewports cannot be focused / activated with the keyboard (possibly also causing issues when using AT).

**Comment by awkawk** [comment]
Response for Scott:
The discussions around this proposed SC have shown that due to the differences between browsers and AT there may be no conclusive way to test if content meets this criterion. 

Also, the requirement seemed mostly covered by 2.1.1 Keyboard in the sense that if a control can be focused and activated with a keyboard, presumably it would also be possible to swipe to it and activate it via double tap it in a mobile screenreader scenario.

Another point is that the issue is also not exclusively related to touch, other input methods may be used concurrenty while AT is turned on. This is the topic in the new [SC 2.5.2 Concurrent Input Mechanisms](https://www.w3.org/TR/2017/WD-WCAG21-20170912/#concurrent-input-mechanisms).

Another new success criterion, [2.5.1 Pointer Gestures](https://www.w3.org/TR/2017/WD-WCAG21-20170912/#pointer-gestures) is meant to close the gap that exists in situations where touch is the input mode and no physical or virtual keyboard alternative is present (a virtual keyboard may not be shown on iOS unless the focus is on some text input). It requires that operation should be possible via single untimed pointer gestures  - i.e. author-defined gestures (swiping to scroll or load content in infinite scroll scenarios, etc) that may not work when AT is turned on would need a fallback way of operation.

A [new note under Conformance requirement 1 Full pages](https://www.w3.org/TR/2017/WD-WCAG21-20170912/#h-note29) explains that a full page includes each variation of the page that is automatically generated by the page for various screen sizes. This would, for example, cover situations where controls in responsive views on narrow viewports cannot be focused / activated with the keyboard (possibly also causing issues when using AT).
